### ISSUE TITLE
Windows Command Prompt: truffle

### ISSUE THREAD

**Issue by NellyWhadsDev** [ConsenSys/truffle#158]
Hello all,

**The issue** I've run into while running truffle on Windows lies in running `truffle build`,`truffle serve`, etc. using the Windows Command Prompt. When I try to do this in a truffle project directory (for example, the one created when using `truffle init`, it simply opens `truffle.js` in the default program associated with JavaScript files ([Visual Studio Code](https://code.visualstudio.com/) for me) and shows a run of errors in the CLI when the program is exited.

**The solution** I am using right now is to just run the desired truffle command in a given truffle project directory using Git Bash ([installed with git](https://git-scm.com/downloads) for Windows). It seems to run without a hitch. Does anyone have a more elegant fix for the issue?

Many thanks,
Nelly

**Comment by tcoulter** [comment]
@NellyWhadsDev What happens when you rename `truffle.js` to `truffle.json` and convert your config file to JSON instead of Javascript? Does that make any noticeable change?

**Comment by ivanchaz** [comment]
@tcoulter Hi, your suggestion is correct. This problem also happen on my workstation. And, by changing the file name and the content of file into JSON, the problem seems to be fixed. 

Actually, I have made the changes so when the user invoke `truffle init` they will get the JSON file instead of js file . So, do you think I should make a pull request for this?

**Comment by tcoulter** [comment]
No, the JS file provides significant features in that it's Javascript instead of JSON. You should try installing Git for Windows which comes with a new terminal called Git Bash which won't have the issue. https://git-for-windows.github.io/

**Comment by tcoulter** [comment]
Whoops, didn't realize you were already using it. As far as a more elegant solution, I think "fixing" the Windows command line is the most elegant, IMO. :)

**Comment by ivanchaz** [comment]
haha, well, if the file is important; then, I guess I will lookup for a better alternative and discuss that with you here.

**Comment by mm-nv** [comment]
Try these workarounds.  (I'm on a windows 7 machine).
1. Be explicit. use truffle.cmd instead of truffle.  e.g. truffle.cmd version
2. Remove ".JS" from the PATHEXT system var.   Based on my experiments, if you dont specify an extension (.xyz) on the command line, windows will append extensions found in the PATHEXT var until it finds a match, first in the current dir, then on the PATH.  When windows finds a match, the default program associated with .js extension is executed.  Removing .js from PATHEXT will prevent windows from thinking that .js files are executable.

**Comment by tcoulter** [comment]
This has been fixed on the master branch of truffle. A release will be made tomorrow containing the fix. The README has been updated in two places that tells you how you can avoid this error:

http://truffle.readthedocs.io/en/latest/getting_started/installation/#recommendations-for-windows
http://truffle.readthedocs.io/en/latest/advanced/configuration/#resolving-naming-conflicts-on-windows

Cheers!

**Comment by aanchi13** [comment]
Hi, I'm getting the same problem and the links provided above doesn't exist. Any help?

**Comment by scarboroughventures** [comment]
Just popping in to say this problem remains unresolved.

**Comment by bisratyalew** [comment]
Hi all, 
I am trying to build my first dApp. I am using windows 8 with Nodejs 6.11.1... I have installed truffle & testrpc properly but when i try to initialize truffle with "truffle init"... it didn't provide me MetaCoin.sol, ConvertLib.sol and others. why is this happening? Here's the screenshot
![trufflegenerated](https://user-images.githubusercontent.com/30002154/35218728-8b7829d4-ff24-11e7-8987-7bf85e373551.jpg)

**Comment by SSPgg** [comment]
if you are using truffle 3, Use "truffle init webpack"
If using truffle 4, you have to use "truffle unbox webpack"
Once project is set , truffle command gets confused with 'truffle.cmd" and the confg file 'truffle.js". Solution is to either give  "truffle.cmd" or change in npm folder to something else like "truffle1"

**Comment by bisratyalew** [comment]
Okay @SSPgg, Thanks for the response, these helped a lot but now I have faced a new problem. I try to create a HelloWorld(helloworld.sol) contract with these sample program...

`pragma solidity ^0.4.4;

contract HelloWorld
{
	function SayHello() public returns (string)
	{
		return ("HelloWorld!");
	}
}`  
and then on the "2_deploy_contracts.js" I required it and deployed it... Here's the screenshot
![deployhwsol](https://user-images.githubusercontent.com/30002154/35270503-3317aaa6-ffe4-11e7-9a79-191648e0ea1e.jpg)

next I started testrpc, but later when I try to compile it I have found these error. 
![truferror](https://user-images.githubusercontent.com/30002154/35270752-e2d7c688-ffe4-11e7-9b57-ed74e8712af1.jpg)

How can i fix this?
